src/ImageSharp/Formats/Jpeg/Components/Decoder/ColorConverters/JpegColorConverter.FromCmyk.cs

```diff
@@ -10,8 +10,8 @@ internal abstract partial class JpegColorConverter
     {
         internal class FromCmyk : JpegColorConverter
         {
-            public FromCmyk()
-                : base(JpegColorSpace.Cmyk)
+            public FromCmyk(int precision)
```

Comment:
I prefer to have more type safety in the codebase. (`-1`, `17` and `222` are invalid values for precision ;) ). I suggest an enum having the following form, so you can easily cast it to `int` when necessary.

```C#
enum JpegPrecision
{
   EightBits = 8,
   TwelveBits = 12
}
```

Reply:
Hmmm... Read section 4.7 here. Lossless jpeg supports additional precision so we should bear that in mind when naming our enum. https://www.w3.org/Graphics/JPEG/itu-t81.pdf

> For DCT-based processes, two alternative sample precisions are specified: either 8 bits or 12 bits per sample. Applications which use samples with other precisions can use either 8-bit or 12-bit precision by shifting their source image samples appropriately. The baseline process uses only 8-bit precision. DCT-based implementations which handle 12-bit source image samples are likely to need greater computational resources than those which handle only 8-bit source images. Consequently in this Specification separate normative requirements are defined for 8-bit and 12-bit DCT-based processes.
> 
> For lossless processes the sample precision is specified to be from 2 to 16 bits

Reply:
Oops. The options I see:
1. Keep the type as is (`int`), and introduce some `DebugGuard` checks in the code
2. Introduce an enum defining all entries for 2-16
3. Since we do not support lossless jpeg, define only `EightBits` and `TwelveBits` for now, and YAGNI out the rest, letting our future selves to find out how to manage the lossless case.

I'm pretty much for option 3.